### PR TITLE
BAU: Improve cold start time of Java lambda

### PR DESCRIPTION
**Pull request by chrisclayson**
## What?

- Add the `JAVA_TOOLS_OPTIONS` environment variable as discussed in
this article:
https://aws.amazon.com/blogs/compute/optimizing-aws-lambda-function-performance-for-java/

## Why?

To improve cold start times of Java lambdas

### CODE REVIEW

**Review by sanisc — APPROVED** [review]

**Review by mrwilson — CHANGES_REQUESTED** [review]
There is a typo, I think it should be `-XX:+TieredCompilation` rather than `-XXx:+TieredCompilation`

File: ci/terraform/account-management/authorizer-warmer.tf [hunk 1]
```diff
@@ -62,6 +62,7 @@ resource "aws_lambda_function" "warmer_function" {
       LAMBDA_QUALIFIER       = aws_lambda_alias.authorizer_alias.name
       LAMBDA_TYPE            = "AUTHORIZER"
       LAMBDA_MIN_CONCURRENCY = var.lambda_min_concurrency
+      JAVA_TOOL_OPTIONS      = "-XXx:+TieredCompilation -XX:TieredStopAtLevel=1"
```
Comment (philandstuff) [inline ci/terraform/account-management/authorizer-warmer.tf:65]:
```suggestion
      JAVA_TOOL_OPTIONS      = "-XX:+TieredCompilation -XX:TieredStopAtLevel=1"
```

File: ci/terraform/account-management/authorizer.tf [hunk 2]
```diff
@@ -35,6 +35,7 @@ resource "aws_lambda_function" "authorizer" {
     variables = {
       TOKEN_SIGNING_KEY_ALIAS = data.aws_kms_key.id_token_public_key.key_id
       ENVIRONMENT             = var.environment
+      JAVA_TOOL_OPTIONS       = "-XXx:+TieredCompilation -XX:TieredStopAtLevel=1"
```
Comment (philandstuff) [inline ci/terraform/account-management/authorizer.tf:38]:
```suggestion
      JAVA_TOOL_OPTIONS       = "-XX:+TieredCompilation -XX:TieredStopAtLevel=1"
```

File: ci/terraform/account-management/sqs.tf [hunk 3]
```diff
@@ -184,6 +184,7 @@ resource "aws_lambda_function" "email_sqs_lambda" {
       CONTACT_US_LINK_ROUTE = var.contact_us_link_route
       NOTIFY_API_KEY        = var.notify_api_key
       NOTIFY_URL            = var.notify_url
+      JAVA_TOOL_OPTIONS     = "-XXx:+TieredCompilation -XX:TieredStopAtLevel=1"
```
Comment (philandstuff) [inline ci/terraform/account-management/sqs.tf:187]:
```suggestion
      JAVA_TOOL_OPTIONS     = "-XX:+TieredCompilation -XX:TieredStopAtLevel=1"
```

File: ci/terraform/modules/endpoint-module/lambda.tf [hunk 4]
```diff
@@ -22,7 +22,8 @@ resource "aws_lambda_function" "endpoint_lambda" {
   }
   environment {
     variables = merge(var.handler_environment_variables, {
-      WARMER_DELAY = var.warmer_delay_millis
+      WARMER_DELAY      = var.warmer_delay_millis
+      JAVA_TOOL_OPTIONS = "-XXx:+TieredCompilation -XX:TieredStopAtLevel=1"
```
Comment (philandstuff) [inline ci/terraform/modules/endpoint-module/lambda.tf:26]:
```suggestion
      JAVA_TOOL_OPTIONS = "-XX:+TieredCompilation -XX:TieredStopAtLevel=1"
```

File: ci/terraform/modules/endpoint-module/warmer.tf [hunk 5]
```diff
@@ -144,8 +144,9 @@ resource "aws_lambda_function" "warmer_function" {
 
   environment {
     variables = merge(var.warmer_handler_environment_variables, {
-      LAMBDA_ARN       = aws_lambda_function.endpoint_lambda.arn
-      LAMBDA_QUALIFIER = aws_lambda_alias.endpoint_lambda.name
+      LAMBDA_ARN        = aws_lambda_function.endpoint_lambda.arn
+      LAMBDA_QUALIFIER  = aws_lambda_alias.endpoint_lambda.name
+      JAVA_TOOL_OPTIONS = "-XXx:+TieredCompilation -XX:TieredStopAtLevel=1"
```
Comment (philandstuff) [inline ci/terraform/modules/endpoint-module/warmer.tf:149]:
```suggestion
      JAVA_TOOL_OPTIONS = "-XX:+TieredCompilation -XX:TieredStopAtLevel=1"
```

File: ci/terraform/oidc/sqs.tf [hunk 6]
```diff
@@ -177,6 +177,7 @@ resource "aws_lambda_function" "email_sqs_lambda" {
       NOTIFY_URL                = var.notify_url
       NOTIFY_TEST_PHONE_NUMBER  = var.notify_test_phone_number
       SMOKETEST_SMS_BUCKET_NAME = local.sms_bucket_name
+      JAVA_TOOL_OPTIONS         = "-XXx:+TieredCompilation -XX:TieredStopAtLevel=1"
```
Comment (philandstuff) [inline ci/terraform/oidc/sqs.tf:180]:
```suggestion
      JAVA_TOOL_OPTIONS         = "-XX:+TieredCompilation -XX:TieredStopAtLevel=1"
```

**Review by philandstuff — APPROVED** [review]

**Review by mrwilson — APPROVED** [review]

**Pull request merged**